networkx/algorithms/community/asyn_lpa.py

```diff
@@ -72,8 +77,8 @@ def asyn_lpa_communities(G, weight=None):
             # Get label frequencies. Depending on the order they are processed
             # in some nodes with be in t and others in t-1, making the
             # algorithm asynchronous.
-            label_freq = Counter({labels[v]: G.edge[v][node][weight]
-                                  if weight else 1 for v in G[node]})
+            label_freq = count_labels([(labels[v], G.edge[v][node][weight]
```

Comment:
How about something like

```python
label_freq = Counter()
for _, v, weight in G.edges(node, weight=weight, default=1):
    label_freq[labels[v]] += weight
```
Basically the same as what you have but a little more concise, and doesn't need to be in its own helper function (it's only a couple lines).